src/core/hub/ShareClassManager.sol

```diff
@@ -79,13 +80,16 @@ contract ShareClassManager is Auth, IShareClassManager {
         auth
     {
         require(exists(poolId, scId_), ShareClassNotFound());
-        require(isIssuance || issuance[poolId][scId_][centrifugeId] >= amount, DecreaseMoreThanIssued());
 
-        uint128 newTotalIssuance =
-            isIssuance ? totalIssuance[poolId][scId_] + amount : totalIssuance[poolId][scId_] - amount;
-        totalIssuance[poolId][scId_] = newTotalIssuance;
-        uint128 newIssuancePerNetwork = isIssuance ? issuance[poolId][scId_][centrifugeId] + amount : issuance[poolId][scId_][centrifugeId] - amount; // forgefmt: disable-line
-        issuance[poolId][scId_][centrifugeId] = newIssuancePerNetwork;
+        IssuancePerNetwork storage ipn = _issuancePerNetwork[poolId][scId_][centrifugeId];
+
+        if (isIssuance) {
+            ipn.issuances += amount;
+            totalIssuance[poolId][scId_] += amount;
```

Comment:
I think we need to swap the order of 
```
shareClassManager.updateShares(originCentrifugeId, poolId, scId, amount, false);
shareClassManager.updateShares(targetCentrifugeId, poolId, scId, amount, true);
```

in `HubHandler`. Since right now, it is decreasing from the total issuance first, and then increasing. Which means you can have an underflow if the transfer is larger than the total submitted shares across networks.

Reply:
good point